docker/contrib/cc/image.bzl

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A rule for creating a C++ Docker image.
+
+The signature of this rule is compatible with cc_binary.
+"""
+
+load(
+    "//docker/contrib/common:lang-image.bzl",
+    "dep_layer",
+    "app_layer",
+)
+load("//docker:pull.bzl", "docker_pull")
+
+def repositories():
+  excludes = native.existing_rules().keys()
+  if "cc_image_base" not in excludes:
+    docker_pull(
+      name = "cc_image_base",
+      registry = "gcr.io",
+      repository = "distroless/cc",
+      # 'latest' circa 2017-07-21
+      digest = "sha256:942eb947818e7e32200950b600cc94d5477b03e0b99bf732b4c1e2bba6eec717",
+    )
+
+def cc_image(name, base=None, deps=[], layers=[], **kwargs):
```

Comment:
I think I prefer the current naming given it's current home under `rules_docker`.  `docker_image` probably would have been a better name :)